webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeCard.js

```diff
@@ -57,6 +57,8 @@ const SystemPurposeCard = ({ hostDetails }) => {
     );
   }
 
+  if (hostIsNotRegistered({ hostDetails })) return null;
```

Comment:
Why not check it at the beginning of the function?

Reply:
@lfu updated :+1:

Reply:
I guess this is why:
```
Error:   48:29  error  React Hook "useSelector" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks
Error:   51:33  error  React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render     react-hooks/rules-of-hooks
```

Reply:
We can move them together.
```
const SystemPurposeCard = ({ hostDetails }) => {
  const hostDetailsStatus = useSelector(selectHostDetailsStatus);
  const [editing, setEditing] = useState(false);
  if (hostIsNotRegistered({ hostDetails })) return null;
```

Reply:
But then we would have logic in the middle of assigning all of the consts. Seems cleaner to just assign all the consts, and then have your `if`s..